scipy/stats/tests/test_multivariate.py

```diff
@@ -1635,6 +1638,187 @@ def test_haar(self):
+        # PDF is correct since this test requires checking the log of that
+        # value.
+        val1 = dist.pdf(x)
+        assert_array_almost_equal(val1, ans)
+        val2 = dist.logpdf(x)
+        assert_array_almost_equal(np.log(val1), val2)
+
+    # https://github.com/scipy/scipy/issues/10042#issuecomment-576795195
+    def test_mvt_with_df_one_is_cauchy(self):
+        x = [9, 7, 4, 1, -3, 9, 0, -3, -1, 3]
+        val = multivariate_t.pdf(x, df=1)
+        ans = cauchy.pdf(x)
+        assert_array_almost_equal(val, ans)
+
+    def test_mvt_with_high_df_is_approx_normal(self):
+        # `normaltest` returns the chi-squared statistic and the associated
+        # p-value. The null hypothesis is that `x` came from a normal
+        # distribution, so a low p-value represents rejecting the null, i.e.
+        # that it is unlikely that `x` came a normal distribution.
+        P_VAL_MIN = 0.1
+
+        dist = multivariate_t(0, 1, df=100000, seed=1)
+        samples = dist.rvs(size=100000)
+        _, p = normaltest(samples)
+        assert (p > P_VAL_MIN)
+
+        dist = multivariate_t([-2, 3], [[10, -1], [-1, 10]], df=100000,
+                              seed=42)
+        samples = dist.rvs(size=100000)
+        _, p = normaltest(samples)
+        assert ((p > P_VAL_MIN).all())
+
+        dist = multivariate_t(0, 1, df=np.inf, seed=7)
+        samples = dist.rvs(size=100000)
+        _, p = normaltest(samples)
+        assert (p > P_VAL_MIN)
+
+    def test_default_arguments(self):
+        dist = multivariate_t()
+        assert_equal(dist.loc, [0])
+        assert_equal(dist.shape, [[1]])
+        assert (dist.df == 1)
+
+    def test_reproducibility(self):
+        rng = np.random.RandomState(4)
+        loc = rng.random(3)
+        tmp = rng.random((20, 3))
+        shape = np.matmul(tmp.T, tmp)
+        dist1 = multivariate_t(loc, shape, df=3, seed=2)
+        dist2 = multivariate_t(loc, shape, df=3, seed=2)
+        samples1 = dist1.rvs(size=10)
+        samples2 = dist2.rvs(size=10)
+        assert_equal(samples1, samples2)
+
+    def test_scalar_list_and_ndarray_arguments(self):
+        dist = multivariate_t(-1, 2, 3)
+        assert(dist.loc == np.array([-1]))
+        assert(dist.shape == np.array([2]))
+        assert(dist.df == 3)
+
+        dist = multivariate_t([-1], [2], 3)
+        assert(dist.loc == np.array([-1]))
+        assert(dist.shape == np.array([2]))
+        assert(dist.df == 3)
+
+        dist = multivariate_t(np.array([-1]), np.array([2]), 3)
+        assert(dist.loc == np.array([-1]))
+        assert(dist.shape == np.array([2]))
+        assert(dist.df == 3)
+
+    def test_allow_singular(self):
+        loc = np.zeros(4)
+        shape = np.eye(4)
+        df = 1
+
+        try:
+            multivariate_t(loc, shape, df, allow_singular=False)
+        except np.linalg.LinAlgError:
+            pytest.fail("Test failed by unexpectedly raising `LinAlgError`.")
```

Comment:
Not sure this is what you want to do here. Is the purpose of this test to test that a `LinAlgError` is raised? 

If so, then usually constructing a matrix explicitly is the preferred way to test the exception raising.

Reply:
Yeah, the first `try/catch` was a mistake. I think this version of the test is pretty clear:

```python
def test_allow_singular(self):
    loc = np.zeros(4)
    shape = np.eye(4)
    df = 1

    # Make shape singular and verify error was raised.
    shape[2, 2] = 0
    allow_singular = False
    seed = 0
    assert_raises(np.linalg.LinAlgError,
                  multivariate_t,
                  loc, shape, df, allow_singular, seed)

    # Check that singular `shape` is allowed when `allow_singular=True`.
    try:
        multivariate_t(loc, shape, df, allow_singular=True)
    except np.linalg.LinAlgError:
        pytest.fail("Test failed by unexpectedly raising `LinAlgError`.")
```

I don't like this test:

```python
val = ss.multivariate_t(loc, shape, df, allow_singular=False).pdf(loc)
assert_almost_equal(val, 0.07599088773175335)
```

because the thing we're asserting is a side-effect of the correct behavior we want to test, namely that `LinAlgError` wasn't raised. Thoughts?

Reply:
Yeah, I see your point. The `_PSD` class is also well-tested in the `multivariate_normal` section. I've removed the second test, so the full unit test is just:

```python
def test_allow_singular(self):
    # Make shape singular and verify error was raised.
    args = dict(loc=[0,0], shape=[[0,0],[0,1]], df=1, allow_singular=False)
    assert_raises(np.linalg.LinAlgError, multivariate_t, **args)
```
The main check of the original unit test is that the `allow_singular` argument is correctly passed through in `__call__`, which is perhaps a minor concern. With this new test, if you don't pass the argument through, all tests still pass.

Reply:
> Yeah, I see your point. The `_PSD` class is also well-tested in the `multivariate_normal` section. I've removed the second test, so the full unit test is just:
> 
> ```python
> def test_allow_singular(self):
>     # Make shape singular and verify error was raised.
>     args = dict(loc=[0,0], shape=[[0,0],[0,1]], df=1, allow_singular=False)
>     assert_raises(np.linalg.LinAlgError, multivariate_t, **args)
> ```
> 
> The main check of the original unit test is that the `allow_singular` argument is correctly passed through in `__call__`, which is perhaps a minor concern. With this new test, if you don't pass the argument through, all tests still pass.

Ok looks good-thanks.